### PR TITLE
[`Phi`] bias should be True

### LINKED ISSUE #35625: The Phi model does not have lm_head bias after upgraded to v4.48.0
### System Info

- `transformers` version: 4.48.0
- Platform: Linux-5.15.0-88-generic-x86_64-with-glibc2.39
- Python version: 3.12.3
- Huggingface_hub version: 0.27.1
- Safetensors version: 0.5.2
- Accelerate version: 1.2.1
- Accelerate config:    not found
- PyTorch version (GPU?): 2.6.0a0+ecf3bae40a (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: <fill in>
- Using GPU in script?: <fill in>
- GPU type: NVIDIA H200

### Who can help?

@ArthurZucker

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

The Phi model's lm_head bias disappear after upgraded to v4.48.0, which is incorrect, see https://github.com/huggingface/transformers/blob/e97d7a5be545d123862d09f6c32e8d90b98bbf0c/src/transformers/models/phi/modeling_phi.py#L727

### Expected behavior

Add lm_head bias back to the Phi model.

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
fixes #35625

### CODE REVIEW

**Comment by ArthurZucker** [comment]
Confirmed that the logits are now good!
